tests/pos/i873.scala

```diff
@@ -0,0 +1,10 @@
+object Test {
+  def call(k: (Int, Int) => Unit): Unit = ???
+  def test = call({ case (x, y) => ()})
+
+  trait X extends Function1[Int, String]
+  implicit def f2x(f: Function1[Int, String]): X = ???
+  ({case _ if "".isEmpty => ""} : X) // allowed, implicit view used to adapt
```

Comment:
I guess this comment is not applicable to dotty. It might be worth leaving a comment that contrasts the approaches, though.